### PR TITLE
Generate inbound shipment

### PR DESCRIPTION
**Pull request by priyankarani**

### CODE REVIEW

File: shipment.py [hunk 1]
```diff
@@ -100,3 +133,216 @@ def export_shipment_status_to_amazon(self):
...
+                (fba_code, inbound_product.quantity)
+            )
+
+        request_items = dict(Member=[{
+            'SellerSKU': sku,
+            'Quantity': str(int(qty)),
+        } for sku, qty in fba_products])
+
+        if not request_items:
+            self.raise_user_error(
```
Comment (tarunbhardwaj) [inline shipment.py]: Add TODO to show which products are not eligible

File: shipment.py [hunk 2]
```diff
@@ -100,3 +133,216 @@ def export_shipment_status_to_amazon(self):
...
+            Button('Cancel', 'end', 'tryton-cancel'),
+            Button('Create', 'create_', 'tryton-go-next', default=True),
+        ]
+    )
+    create_ = StateAction('stock.act_shipment_internal_form')
+
+    def find_product_using_sku(self, sku):
+        Listing = Pool().get('product.product.channel_listing')
+
+        listings = Listing.search([
```
Comment (tarunbhardwaj) [inline shipment.py]: Add channel also in domain

File: channel.py [hunk 3]
```diff
@@ -112,6 +113,18 @@ def get_mws_api(self):
             account_id=self.amazon_merchant_id,
         )
 
+    def get_mws_connection_api(self):
```
Comment (tarunbhardwaj) [inline channel.py]: get_mws_api_boto_connection

**Comment by tarunbhardwaj** [comment]
LGTM

File: shipment.py [hunk 4]
```diff
@@ -6,12 +6,45 @@
...
+]
 __metaclass__ = PoolMeta
 
 
+class StockLocation:
+    __name__ = 'stock.location'
+
+    # This field is added so we can have sku of the product (fullfilled by
+    # amazon) while sending product info to amazon network
+    channel = fields.Many2One(
```
Comment (prakashpp) [inline shipment.py]: I dont think this is correct. Channel already has a foreign key for fba warehouse.

File: channel.py [hunk 5]
```diff
@@ -112,6 +113,18 @@ def get_mws_api(self):
             account_id=self.amazon_merchant_id,
         )
 
+    def get_mws_boto_connection_api(self):
```
Comment (prakashpp) [inline channel.py]: Create a task to use boto api for other operation too

File: shipment.py [hunk 6]
```diff
@@ -6,12 +6,42 @@
...
+    )
+
+    @fields.depends('parent')
+    def on_change_with_fba_channel(self, name=None):
+        Channel = Pool().get('sale.channel')
+
+        if self.parent:
+            channel = Channel.search([
+                ('source', '=', 'amazon_mws'),
+                ('fba_warehouse', '=', self.parent.id),
```
Comment (prakashpp) [inline shipment.py]: this will break if location is warehouse, there will be no parent.

Comment (priyankarani) [inline shipment.py]: I'm checking for parent above

Comment (prakashpp) [inline shipment.py]: ignore

**Comment by priyankarani** [comment]
@prakashpp please review this

**Comment by prakashpp** [comment]
LGTM

**Pull request merged**